[CS2113T-T09-1] MediTracker

kyuichyi opened the pull request:
MediTracker is a Command Line Interface (CLI) app to track your medications. It gives you the capability to manage your medications taken on a daily basis such as indicating if you have taken your medication and when it is going to expire.

Luo-Z-Y commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,126 @@
+
+---
+
+### List Medication Command
```
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/97681546/6e10108a-9b3b-4209-93aa-10d830fe51cf)
Perhaps you could consider only highlighting the code part instead of the whole sentence.

Luo-Z-Y commented on docs/DeveloperGuide.md:
````diff
@@ -4,30 +4,126 @@
+- execute(MedicationManager medicationManager) - Executes the list command and performs its specific task, -t.
+```Where the task can be either 'list -t all' to list all medications or 'list -t today' to list medications for the day, which is divided into three categories -> Morning, Afternoon and Evening. ```
+
+The 'list -t' command requires the following:
````
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/97681546/3b1befb8-9795-4dac-bc6c-bb0188d5a0f9)
You could potentially highlight the code part in this segment to make it more readable.

Luo-Z-Y reviewed (commented):
In general well done, however UML not done.

liuy1103 commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
+The 'search' command requires the following:
+1. To be added.
+
+### Utilising the argument parser
```
Consider using a class diagram to illustrate this to provide visual feedback 😄

liuy1103 reviewed (commented):
Overall LGTM, but looks a little simplistic, perhaps more UML diagrams and illustrations for feature enhancements 😄

ChinYanXu commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
 
+
+## Design & implementation
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
```
Maybe you could remove these documentation instructions as they are not necessary

ChinYanXu commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
+### Export to JSON File
+
+![img.png](images/JsonFileExport.png)
+
```
You could draw a dotted arrow away from FileReaderWriter to represent the output from "List<String> dailyMedicationList = FileReaderWriter.loadDailyMedicationData();" in your MediTracker class at line 101.

ChinYanXu commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
+---
+
+### Export to JSON File
+
```
Consider adding more design sequence diagrams to describe the functions described in the design and implementation

ChinYanXu commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
+
+
+### Simulated Time
+Talk about the advaned feature. Offset based on the system time so that the user or developer does not have to worry about calculating the time and can just type in the time.
```
Consider fixing the typo "advaned" to "advanced" here.

JianJiaT commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+MediTracker ensures that you would now forget your overall schedule on what time and day to take your medication.
```
Shouldn't this be "ensures that you would not forget" instead of "ensures that you would now forget"?

JianJiaT commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Add Medication Command
+The add medication command extends from the Command parent class and implements the following operations:
+- execute(MedicationManager, DailyMedicationManager, Ui) - Adds the medication object into the respective medication managers.
+- setMedicineAttributes() - Sets the medicine object's attributes to be added to the medicine managers. 
+- assertionTest(MedicationManager, DailyMedicationManager) - Asserts that medicine has been added to both medication managers.
+- parseStringToValues(Arguments) - Parses string input for medicine quantity and dosage into double type.
```
Perhaps add sequence diagram or class diagram here to show the interactions and relationships since this seems to involve quite a lot of classes?

JianJiaT commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
+The 'list -t' command requires the following:
+1. 'all' - to run printAllMedications() from the MedicationManager.
+2. 'today' - to run printMedications() from the DailyMedicationManager.
+3. 'today -m' to run printTodayMedications(List<Medication> medications, List<DailyMedication> subList, String period)
+   from the DailyMedicationManager
+4. 'today -a' to run printTodayMedications(List<Medication> medications, List<DailyMedication> subList, String period)
+   from the DailyMedicationManager
+5. 'today -e' to run printTodayMedications(List<Medication> medications, List<DailyMedication> subList, String period)
+   from the DailyMedicationManager
```
Perhaps clarify what is the difference between inputting "today -m", "today -a" and "today -e"?

JianJiaT commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
+|v1.0| user              | know what medications i have to take throughout the day      | mark them as takenor not, thus not miss or forget to take what I have to take for the day                                                              |
+|v1.0| user              | edit the details in my medication list                       | ensure that its medication information is up-to-date with my medifications                                                                             |
+|v1.0| user              | delete medications medication list                           | remove medications that I do not need will not be there anymore                                                                                        |
+|v1.0| user              | search medications from existing medicine library            | search medications locally to have a quick preview of them and their purpose without the web                                                           |
+|v1.0| multipurpose user | Store data locally                                           | Use the application and see the data even when offline between sessions                                                                                |
+|v1.0| user              | know the list of medications I have added                    | have a quick overview of the medication list and check the quantity and expiry date of each medication                                                 |
```
Perhaps clarify what is a "multipurpose user" and how it is different from a normal user?

JianJiaT commented on docs/DeveloperGuide.md:
````diff
@@ -4,30 +4,136 @@
+- execute(MedicationManager medicationManager) - Executes the list command and performs its specific task, -t.
+```Where the task can be either 'list -t all' to list all medications or 'list -t today' to list medications for the day, which is divided into three categories -> Morning, Afternoon and Evening. ```
+
+The 'list -t' command requires the following:
````
Perhaps specify that the following parameters need to follow the "list -t" command instead of preceding it or some other order?

JianJiaT commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
```
Perhaps add a table of contents at the top of the DG?

JianJiaT commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
+|v1.0| user              | search medications from existing medicine library            | search medications locally to have a quick preview of them and their purpose without the web                                                           |
+|v1.0| multipurpose user | Store data locally                                           | Use the application and see the data even when offline between sessions                                                                                |
+|v1.0| user              | know the list of medications I have added                    | have a quick overview of the medication list and check the quantity and expiry date of each medication                                                 |
+|v2.0| user              | find a to-do item by name                                    | locate a to-do without having to go through the entire list                                                                                            |
```
Perhaps remove this user story since it seems to be a remnant from the project template that does not seem to be relevant to this application?

JianJiaT reviewed (commented):
Good start but could use more UML diagrams and fix some issues

Xb990219 commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
+tests have been added to ensure that certain attributes of the `Argument` classes do not clash with
+one another. `HelpArgument` is automatically added when an `ArgumentList` object is instantiated.
+Calling `ArgumentList.parse` with empty `String` or invoking with the argument `-h` will trigger the
+help message to be printed on the console.
```
Consider explaining how "trigger" works.

Xb990219 commented on docs/DeveloperGuide.md:
```diff
@@ -4,30 +4,136 @@
+the key. In order to query the returned data structure, you can utilise the same enum.
+
+In order to utilise the argument parser,
+1. Determine if the `Argument` variant already exist. If not, create a new class and extend the `Argument` class.
```
Good explaination

e0958902 committed:
Breakpoint for Add and List command

royaliceblue committed:
docs: Add Javadoc to readCommand() method in Ui

T0nyLin committed:
Adds screenshot for Java11 to use in UG

e0958902 committed:
Remove spaces and adjust alignments

e0958902 committed:
Remove spaces and adjust alignments v0.01

e0958902 committed:
Add spaces after : or colon in Table of Contents

royaliceblue committed:
docs: Update Javadoc for readCommand method

e0958902 committed:
Remove spaces and adjust alignments v0.02

e0958902 committed:
Break v0.04

e0958902 committed:
Remove spaces and adjust alignments v0.03

e0958902 committed:
Remove spaces and adjust alignments v0.04

e0958902 committed:
Remove spaces and adjust alignments v0.05

e0958902 committed:
Remove spaces and adjust alignments v0.06

e0958902 committed:
Breakpoint for View and Update command

e0958902 committed:
Update ordering and expiry dates

e0958902 committed:
Update tip box

e0958902 committed:
Breakpoint for Search, Help, save, and exit command

e0958902 committed:
Breakpoint for FAQ and Glossary

e0958902 committed:
Merge pull request #232 from nickczh/nickczh-fix-Ctrl-C

fix: Handle NoSuchElementException when Ctrl-C termination

T0nyLin committed:
Removes unnecessary lines in DailyMedicationManagerTest

T0nyLin committed:
Adds Junit tests for errornous list commands

T0nyLin committed:
Merge branch 'master' into ApplyPatch

e0958902 committed:
Merge branch 'master' into UG-returnButton

T0nyLin committed:
Replaces * in imports and reorders instance variable definition

T0nyLin committed:
Removes ss for Java11

T0nyLin committed:
Merge pull request #230 from T0nyLin/ApplyPatch

Include more data for DailyMedication storage to textfile

e0958902 committed:
Merge branch 'master' into UG-breakpoints

# Conflicts:
#	docs/UserGuide.md

e0958902 committed:
Update PPP

e0958902 committed:
Fix exit broken link

wenenhoe committed:
Merge branch 'master' into wenen-RefactorArgumentException

# Conflicts:
#	src/main/java/meditracker/command/ListCommand.java
#	src/test/java/meditracker/command/UntakeCommandTest.java

wenenhoe committed:
[#123] Simplify argument exception (#229)

e0958902 committed:
Update ## Quick Start in UG

e0958902 committed:
Update Quick Start in UG

e0958902 committed:
Update User Guide add command formatting

e0958902 committed:
Change REMARKS to lowercase

e0958902 committed:
Remove extra line in add command

e0958902 committed:
Attempt to fix alignment

e0958902 committed:
Remove extra space

e0958902 committed:
Merge branch 'master' into UG-breakpoints

e0958902 committed:
Add MediTracker image to User Guide

e0958902 committed:
Update add command to only remarks is optional

e0958902 committed:
Update User Guide Daily Medications

wenenhoe committed:
fix(bug): Add optional field into parsedArguments

wenenhoe committed:
fix(bug): Fix wrong sequence of name change

Medication name has to be updated after DailyMedication name changes

royaliceblue committed:
docs: Link ToC to items and back, add command summary

Let's
* Include a snapshot under the Quick Start section

e0958902 committed:
Update Help command to follow expected output

royaliceblue committed:
docs: Update PPP

wenenhoe committed:
fix(bug): Fix aborted modify

But name has not changed yet

royaliceblue committed:
Merge branch 'master' into nickczh-UG-Improvements

royaliceblue committed:
docs: Update PPP

royaliceblue committed:
Merge branch 'nickczh-UG-Improvements' of https://github.com/nickczh/tp into nickczh-UG-Improvements

royaliceblue committed:
docs: Update PPP

royaliceblue committed:
Merge pull request #236 from nickczh/nickczh-UG-Improvements

docs: UG Improvements to Table of Contents, Command Summary and more

e0958902 committed:
Merge branch 'master' into UG-breakpoints

# Conflicts:
#	docs/UserGuide.md

e0958902 committed:
Fix issue on linking

e0958902 committed:
Fix link issue v 0.01

e0958902 committed:
Add Windows Java 11 in User Guide

e0958902 committed:
Update Windows image

wenenhoe committed:
[#225] Add breakpoints, adjust alignments and improve on consistency (#234)

wenenhoe committed:
chore: Reorder arguments to be same as add

wenenhoe committed:
chore: Rename title

wenenhoe committed:
fix(docs): Change repeat to mandatory

wenenhoe committed:
docs: Reorder and add warning for modify

wenenhoe committed:
Revert "chore: Rename title"

This reverts commit d1724de8b960e30e4ed413954eb0b165b16ed07d.

wenenhoe committed:
docs: Remove unnecessary lines

wenenhoe committed:
docs: Reorganise headings

wenenhoe committed:
docs: Rename heading

wenenhoe committed:
docs: Add missing help param

wenenhoe committed:
fix(docs): Fix command summary formatting

Code blocks not displaying correctly as 1 code block used for multiple

wenenhoe committed:
fix(docs): Change search command format

wenenhoe committed:
fix(docs): Fix command summary formatting

wenenhoe committed:
docs: Change untake in command summary

wenenhoe committed:
docs: Try new layout

wenenhoe committed:
Revert "docs: Try new layout"

This reverts commit f43f120ef488625ec00a49e95cdef2ab3da2c666.

wenenhoe committed:
docs: Add admonition box

wenenhoe committed:
fix(docs): Change href

wenenhoe committed:
docs: Add tips on no units for quantity dosages

wenenhoe committed:
docs: Combine into single tip box

wenenhoe committed:
docs: Add danger box for delete

wenenhoe committed:
docs: Add bottom navbar

wenenhoe committed:
docs: Restore sections

wenenhoe committed:
docs: Update contributions

kyuichyi committed:
Merge branch 'master' into Update-UG-Search

kyuichyi committed:
docs: Update UserGuide.md to fit format

kyuichyi committed:
docs: update kyuichyi.md

annoy-o-mus committed:
style: fix style

annoy-o-mus committed:
[#189] docs: Fix up docs and add more info for modify (#237)

annoy-o-mus committed:
Merge branch 'master' into 146-styling-checks

e0958902 committed:
Merge branch 'master' into eric-checkstyle

e0958902 committed:
style: fix style in ListCommand

e0958902 committed:
style: fix style in ListCommandTest

e0958902 committed:
style: fix style in ViewCommand

e0958902 committed:
style: fix style in ViewCommandTest

e0958902 committed:
style: fix style in MedicationManager

e0958902 committed:
style: fix style in MedicationManagerTest

e0958902 committed:
style: fix style in Medication

e0958902 committed:
Merge pull request #239 from annoy-o-mus/146-styling-checks

[#146] style: fix style

e0958902 committed:
Merge branch 'master' into eric-checkstyle

e0958902 committed:
Undo Checks on methods not assigned to me

e0958902 committed:
Remove blank line in MedicationManagerTest

e0958902 committed:
Remove blank line and shift author tag in MedicationManagerTest

annoy-o-mus committed:
[#148] style: fix style (#240)

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

annoy-o-mus committed:
test: Use some valid values for save file

annoy-o-mus committed:
refactor: Shift to `storage.MediTrackerFileConfig`

e0958902 committed:
Create ViewCommand sequence diagram

e0958902 committed:
Add ViewCommand sequence diagram to docs/images/

e0958902 committed:
Add View Command to Developer Guide

wenenhoe committed:
test: Add modify tests

annoy-o-mus committed:
Refactor FileReaderWriter

Also a fixes a minor bug relating to the missing clearing of
DailyMedicationManager for loading operations

wenenhoe committed:
docs: Add javadoc for Ui show Message methods

annoy-o-mus committed:
Refactor FilePathChecker

annoy-o-mus committed:
Refactor JsonImporter

wenenhoe committed:
fix: Fix style for command classes

wenenhoe committed:
fix: Fix style for exception classes

wenenhoe committed:
fix: Fix styles

annoy-o-mus committed:
Refactor save and load commands

wenenhoe committed:
fix: Fix style for argument classes

annoy-o-mus committed:
Refactor MediTrackerTime

wenenhoe committed:
fix: Fix styles for tests

kyuichyi committed:
Merge branch 'master' into Update-UG-Search

annoy-o-mus committed:
Refactor various files

annoy-o-mus committed:
style: Fix import order

wenenhoe committed:
[#244, #245] Add modify tests and fix bugs (#243)

kyuichyi committed:
Merge branch 'master' into Update-UG-Search

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

royaliceblue committed:
refactor: Adhere code quality to strict checkstyle

royaliceblue committed:
docs: Update PPP

T0nyLin committed:
Improve checkstyle

royaliceblue committed:
Merge branch 'master' into nickczh-CodeQuality-V2.1

e0958902 committed:
Update Project Portfolio Page

e0958902 committed:
Improve on sentence structure

royaliceblue committed:
refactor: Fix import order

royaliceblue committed:
Merge pull request #248 from nickczh/nickczh-CodeQuality-V2.1

refactor: Enhance code quality for stricter style checks

annoy-o-mus committed:
Merge branch 'master' into 241-fix-code-quality

kyuichyi committed:
Merge branch 'master' into Update-UG-Search

kyuichyi committed:
docs: update UG to include note for user

e0958902 committed:
Update Sequence Diagram

e0958902 committed:
Update Developer Guide v0.1

T0nyLin committed:
Adds a SD for addToSubLists() in DailyMedicationManager

kyuichyi committed:
docs: grammar edits

e0958902 committed:
Merge pull request #238 from kyuichyi/Update-UG-Search

docs: Update search in UG

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

e0958902 committed:
Update typo in User Guide

e0958902 committed:
Update User Guide symbols

annoy-o-mus committed:
[#241] fix code quality (#247)

e0958902 committed:
Add explanation steps for View Command sequence diagram

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

e0958902 committed:
Merge pull request #242 from e0958902/eric-updateDeveloperGuide

Add View Command Sequence Diagram to Developer Guide

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

annoy-o-mus committed:
docs: Update PPP

T0nyLin committed:
Adds the Glossary section in UG

annoy-o-mus committed:
docs: fix PPP typo

T0nyLin committed:
Merge branch 'master' into ImproveDGnUG

wenenhoe committed:
fix: Fix style again

wenenhoe committed:
Merge branch 'master' into wenen-FixStyle

# Conflicts:
#	src/main/java/meditracker/command/ModifyCommand.java

e0958902 committed:
Update PPP v0.1

e0958902 committed:
Update PPP v0.2

annoy-o-mus committed:
docs: Update DG Manual Testing

annoy-o-mus committed:
docs: Fix DG format

annoy-o-mus committed:
docs: Add space

e0958902 committed:
Update PPP v0.3

e0958902 committed:
Update PPP v0.4

e0958902 committed:
Update PPP v0.5

T0nyLin committed:
Adds a SD for ListCommand

annoy-o-mus committed:
docs: Update user stories and NFR

e0958902 committed:
Update DG Sequence Diagram

annoy-o-mus committed:
docs: Fix broken link

T0nyLin committed:
Adds SD to DG

T0nyLin committed:
Updates SD for AddCommand

T0nyLin committed:
Removes used User in puml

wenenhoe committed:
fix: Fix style again

T0nyLin committed:
Adds a ToC in DG

annoy-o-mus committed:
[#151] Fix style (#246)

annoy-o-mus committed:
docs: Update DG: Getting started

annoy-o-mus committed:
docs: Update headings

T0nyLin committed:
Updates user stories in DG

e0958902 committed:
Update DG Sequence Diagram

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

T0nyLin committed:
Updates ListCommand uml

e0958902 committed:
Update PPP v0.6

annoy-o-mus committed:
docs: Update Sequence Diagram for FileExport

e0958902 committed:
DG test v0.01

T0nyLin committed:
Adds a glossary for DG

T0nyLin committed:
Updates glossary in DG

annoy-o-mus committed:
docs: Add import sequence diagram

T0nyLin committed:
Fixes typo

T0nyLin committed:
Merge branch 'master' into ImproveDGnUG

e0958902 committed:
DG test v0.02

e0958902 committed:
DG test v0.03

e0958902 committed:
Update UG arrow

e0958902 committed:
Update DG v0.3

e0958902 committed:
Update DG v0.4

e0958902 committed:
Update DG v0.5

e0958902 committed:
Update PPP v0.6

wenenhoe committed:
docs: Add sequence diagrams for `ModifyCommand`

wenenhoe committed:
docs: Add docs on ModifyCommand for DG

annoy-o-mus committed:
docs: Update DG

annoy-o-mus committed:
Merge branch 'master' into docs-dg-ppp

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/team/annoy-o-mus.md

e0958902 committed:
Merge pull request #249 from T0nyLin/ImproveDGnUG

Improve DG and UG

e0958902 committed:
Merge pull request #250 from e0958902/eric-updateDeveloperGuide

Update developer guide and project portfolio page

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

annoy-o-mus committed:
Merge branch 'master' into docs-dg-ppp

# Conflicts:
#	docs/DeveloperGuide.md

e0958902 committed:
Merge pull request #252 from annoy-o-mus/docs-dg-ppp

[#251] docs: DG and PPP

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

annoy-o-mus committed:
docs: add Q&A

e0958902 committed:
Add page break in DG v0.01

annoy-o-mus committed:
docs: Update UG

wenenhoe committed:
docs: Add class diagram for argument package

wenenhoe committed:
docs: Add diagram into DG

e0958902 committed:
Add page break in DG v0.02

wenenhoe committed:
docs: Add contributions to PPP

e0958902 committed:
Add page break in DG v0.03

annoy-o-mus committed:
docs: Update TOC

e0958902 committed:
Add page break in DG v0.04

e0958902 committed:
Add page break in DG v0.05

wenenhoe committed:
Merge branch 'master' into wenen-AddContentToDG

# Conflicts:
#	docs/DeveloperGuide.md

e0958902 committed:
Add page break in DG v0.06

wenenhoe committed:
fix: Merge conflict residues

annoy-o-mus committed:
docs: Edit save file

e0958902 committed:
Add page break in DG v0.07

royaliceblue committed:
docs: Update DG and PPP

e0958902 committed:
PPP

annoy-o-mus committed:
docs: Update UG

wenenhoe committed:
docs: Add modifycommand to ToC

royaliceblue committed:
fix: Indentation

annoy-o-mus committed:
docs: update UG

annoy-o-mus committed:
docs: Update

wenenhoe committed:
Merge pull request #253 from wenenhoe/wenen-AddContentToDG

docs: Add content to DG

royaliceblue committed:
Merge branch 'master' into nickczh-DG-V2.1

royaliceblue committed:
Merge pull request #254 from nickczh/nickczh-DG-V2.1

docs: Update DG and PPP

e0958902 committed:
PPP Update

annoy-o-mus committed:
Merge remote-tracking branch 'teamOrg/master' into 255-docs-ug-final

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

# Conflicts:
#	docs/DeveloperGuide.md

annoy-o-mus committed:
Merge remote-tracking branch 'teamOrg/master' into 255-docs-ug-final

annoy-o-mus committed:
[#255] docs: UG (#257)

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

T0nyLin committed:
Updates PPP

royaliceblue committed:
Merge branch 'master' into ImprovePPP

e0958902 committed:
Merge pull request #256 from e0958902/eric-updateDeveloperGuide

Update developer guide for page breaks and user stories ordering

kyuichyi committed:
docs: Update search DG

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

kyuichyi committed:
Merge branch 'master' into Search-DGdocs

T0nyLin committed:
Merge pull request #259 from T0nyLin/ImprovePPP

Updates PPP

e0958902 committed:
Update DG

kyuichyi committed:
Merge branch 'master' into Search-DGdocs

kyuichyi committed:
Update DeveloperGuide.md

e0958902 committed:
Merge branch 'master' into eric-updateDeveloperGuide

e0958902 committed:
Merge pull request #260 from kyuichyi/Search-DGdocs

docs: Update search DG

T0nyLin committed:
Merge pull request #261 from e0958902/eric-updateDeveloperGuide

Eric update developer guide
